.github/workflows/ci.yml

```
@@ -36,8 +36,9 @@ jobs:
     - uses: actions/checkout@v4
     - name: install clang repo
       run: |
+        ubuntu_codename=`LC_ALL=C sed 's/^ *UBUNTU_CODENAME= *\([a-z]*\).*$/\1/p; d' /etc/os-release`
```

Why match only whitespace before, but any character after? Also why no whitespace before the equals sign?
```suggestion
        ubuntu_codename=`LC_ALL=C sed 's/^ *UBUNTU_CODENAME *= *\([a-z]*\) *$/\1/p; d' /etc/os-release`
```

---

> Why match only whitespace before, but any character after? Also why no whitespace before the equals sign?

The "any character after" is to discard any potential comment text there, `# like this`. It's not an error.

I corrected the whitespace before the equal sign, though.

---

I this case be explicit about optionally matching a comment, like ` *\(#.*\)?$` (untested) …

---

Better fail to detect the codename in a strangely formatted os-release file, than to execute arbitrary code on a CI system … ;-)

---

There is no arbitrary code execution at all. :) I could have just used the `source` (or `.`) command on the `os-release` file (as many lazy shell scripts do) and let arbitrary commands to be injected in. This regex has been stricter than the original `os-release` format already. For example, it doesn't recognise quoted value (`UBUNTU_CODENAME="noble"`).

---

ACK. And I'm fine with being strict here.

And my comment should be read as: I'd rather have the CI fail to detect the version here than that there is any chance something strange in this place allows to execute some code to execute unintentionally.